cogs/tags/tags.py

```diff
@@ -271,33 +283,65 @@ def verify_lookup(self, lookup):
         if len(lookup) > 100:
             raise RuntimeError("Tag name is a maximum of 100 characters.")
 
-    @tag.command(name="max")
+    @tag.group("settings")
     @commands.guild_only()
     @checks.mod_or_permissions()
-    async def max(self, ctx: Context, num_tags: int = None):
-        """Set the max number of tags per user. Leave blank to show current setting.
+    async def settings(self, ctx: Context):
+        """Tag settings."""
+
+    @settings.command(name="max")
+    async def max(self, ctx: Context, role: discord.Role, num_tags: int):
+        """Set the max number of tags per member per role.
+
+        For each member of the specified role, each member will have a maximum
+        number of tags they can create. If the member is part of more than one
+        role, then they will take the MAXIMUM number from the roles that they
+        have.
 
         This limit does not apply to admins or mods.
 
         Parameters:
         -----------
+        role: discord.Role
+            The role to set a maximum for.
         num_tags: int
-            The maximum number of tags per user.
+            The maximum number of tags per member for that role.
+            If 0, then the tier is removed.
         """
-        if not num_tags:
-            limit = self.settings.get(KEY_MAX, DEFAULT_MAX)
-            await ctx.send("The current tag limit per user is {}.".format(limit))
-            return
         if num_tags < 0:
             await ctx.send("Please set a value greater than 0.")
             return
 
-        await self.settings.put(KEY_MAX, num_tags)
-        await ctx.send("The tag limit was set to {}".format(num_tags))
+        async with self.configV3.guild(ctx.guild).tiers() as tiers:
+            if num_tags == 0:
+                if str(role.id) in tiers.keys():
+                    del tiers[str(role.id)]
+                await ctx.send(f"{role.name} will not be allowed to add tags")
+            else:
+                tiers[role.id] = num_tags
+                await ctx.send(f"The tag limit for {role.name} was set to {num_tags}.")
+
+    @settings.command(name="tiers")
+    async def tiers(self, ctx: Context):
+        """Show the tiers and their respective max tags."""
+        tiers = await self.configV3.guild(ctx.guild).tiers()
+        validTiers = []
+        msg = ""
+        for roleId, maxTags in tiers.items():
+            role = discord.utils.get(ctx.guild.roles, id=int(roleId))
+            if not role:
+                continue
+            validTiers.append((role, maxTags))
+        if validTiers:
+            msg = "Below is a list of roles and their tag limits:\n"
+            validTiers.sort(key=lambda x: x[1])
+            for role, maxTags in validTiers:
+                msg += f"{role.name}: {maxTags}\n"
```

Comment:
Minor:
Add the strings to a list and then set msg to the concatenated list after the loop
```
msg = "".join(msg_list)
```

Reply:
Sure, fixed.